default/python/AI/ProductionAI.py

```diff
@@ -97,10 +97,15 @@ def cur_best_military_design_rating():
         return _best_military_design_rating_cache[current_turn]
     priority = PriorityType.PRODUCTION_MILITARY
     if priority in _design_cache:
+        # TODO: rework the below code, CombatRatingsAI and possibly  WarShipDesigner so that the below uses a rating
+        # provided by CombatRatingsAI (so as to be more suitable for its actual use and to avoid making any assumptions
+        # about how the WarShipDesigner may apply cost adjustments to its rating).
```

Comment:
> It could be that optimize_design could save two ratings values for the design cache-- in addition to the current one that is adjusted for cost and other considerations, it could save a 'rated_strength' value, which for warships would be the rating a la CombatRatingsAI, for troop ships would be the effective number of troops, etc.

Sounds reasonable.